### PR TITLE
lsix: Fix Font Issues

### PR DESCRIPTION
**Pull request by justinlime**
## Description of changes

Fixes the error:
montage: unable to read font `(null)' @ error/annotate.c/RenderFreeType/1741

This is caused by imagemagick not being built with ghostscriptSupport

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by jtbx — CHANGES_REQUESTED** [review]
Looks fine but you should make a separate commit for switching to `finalAttrs`.

File: pkgs/tools/graphics/lsix/default.nix [hunk 1]
```diff
@@ -1,17 +1,17 @@
-{ lib, stdenvNoCC, fetchFromGitHub, makeWrapper, imagemagick }:
+{ lib, stdenv, fetchFromGitHub, makeBinaryWrapper, imagemagick }:
 
-stdenvNoCC.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
```
[the remark this answers is not in the document]
Comment (justinlime) [reply pkgs/tools/graphics/lsix/default.nix]: Amended the changes you requested

**Review by jtbx — APPROVED** [review]

File: pkgs/tools/graphics/lsix/default.nix [hunk 2]
```diff
@@ -20,18 +20,21 @@ stdenvNoCC.mkDerivation rec {
 
     runHook postInstall
   '';
+  wrapperPath = lib.makeBinPath ([
+    (imagemagick.override { ghostscriptSupport = true;})
+  ]);
```
Comment (fpletz) [inline pkgs/tools/graphics/lsix/default.nix:25]: This doesn't need to be an a parameter for `mkDerivation`, right? Either just `let` this in `postFixup` or interpolate the `makeBinPath` call directly into the string as before.

Comment (justinlime) [reply pkgs/tools/graphics/lsix/default.nix:25]: This allows it to be overwritten by the finalAttrs pattern

File: pkgs/tools/graphics/lsix/default.nix [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
-stdenvNoCC.mkDerivation rec {
+stdenvNoCC.mkDerivation (finalAttrs: {
   pname = "lsix";
   version = "1.8.2";
 
   src = fetchFromGitHub {
     owner = "hackerb9";
-    repo = pname;
-    rev = version;
+    repo = finalAttrs.pname;
```
Comment (eclairevoyant) [inline pkgs/tools/graphics/lsix/default.nix:9]: This pattern should never be used, `pname` and `repo` are independent.

```suggestion
    repo = "lsix";
```

File: pkgs/tools/graphics/lsix/default.nix (hunk 2 above)
Comment (eclairevoyant) [reply pkgs/tools/graphics/lsix/default.nix:25]: `postFixup` can be overridden as well, I'd rather not use bespoke patterns without a solid reason

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/tools/graphics/lsix/default.nix (hunk 3 above)
Comment (justinlime) [reply pkgs/tools/graphics/lsix/default.nix]: I believe i've fixed the changes you requested

**Review by nrabulinski — APPROVED** [review]

**Comment by eclairevoyant** [comment]
Do you have a way to reproduce the original issue? I don't encounter any error/crash when I run `lsix` in some random dir.

**Comment by justinlime** [comment]
> Do you have a way to reproduce the original issue? I don't encounter any error/crash when I run `lsix` (without this change) in some random dir.

It could possibly be dependent on the terminal emulator. It requires a Sixel supported terminal to use. I am using the Foot terminal on Wayland with Sixel enabled. When used in a directory with valid images, it should output the results of all images in the directory with a preview along with the files' name below it.

Heres the output I get when using the current version:


When using the updated version from this PR, here are the results:


If you aren't able to reproduce on your side, I will attempt to reproduce in a Virtual Machine.

**Review by eclairevoyant — APPROVED** [review]
I didn't get any error before the patch, but after the patch it seems to choose a more appropriate font, so that's a net improvement.

**Comment by eclairevoyant** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lsix</li>
  </ul>
</details>

**Review by fpletz — APPROVED** [review]

**Pull request merged**